### PR TITLE
[docs] update EAS Submit docs + add page on eas.json

### PR DESCRIPTION
**Pull request by dsokal**
# Why

- https://github.com/expo/eas-cli/pull/555 added support for submit profiles in `eas.json`.
- https://github.com/expo/eas-cli/pull/590 drops support for `eas submit` CLI params. From now on submissions have to be configured with `eas.json`.

# How

- I added a page on configuring `eas submit` with `eas.json`.
- I updated the other EAS Submit docs.

# Test Plan

`yarn dev`

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/build/eas-json.md [hunk 1]
```diff
@@ -67,10 +66,16 @@ Generally, the schema of this file looks like this:
...
     ...
+  },
+  "submit": {
+    // EAS Submit configuration
+    ...
   }
 }
 ```
 
+If you're also using EAS Submit, [see how to use `eas.json` for configuring your submissions](/submit/eas-json.md).
```
Comment (wkozyra95) [inline docs/pages/build/eas-json.md:77]:
```suggestion
If you're also using EAS Submit, [see how to use `eas.json` to configure your submissions](/submit/eas-json.md).
```

File: docs/scripts/schemas/unversioned/eas-json-submit-android-schema.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  },
+  {
+    name: 'track',
+    enum: ['production', 'beta', 'alpha', 'internal'],
+    description: ['The track of the application to use.']
+  },
+  {
+    name: 'releaseStatus',
+    enum: ['completed', 'draft', 'halted', 'inProgress'],
+    description: ['Release status - used when uploading new APKs/AABs.'],
```
Comment (wkozyra95) [inline docs/scripts/schemas/unversioned/eas-json-submit-android-schema.js:15]: I don't understand this description

**Review by wkozyra95 — APPROVED** [review]

File: docs/scripts/schemas/unversioned/eas-json-submit-android-schema.js (hunk 2 above)
Comment (dsokal) [reply docs/scripts/schemas/unversioned/eas-json-submit-android-schema.js:15]: Copy-pasted from CLI :D

**Review by wkozyra95 — APPROVED** [review]

**Review by barthap — APPROVED** [review]

File: docs/constants/navigation.js [hunk 3]
```diff
@@ -36,7 +36,7 @@ const sections = [
     reference: [
       'EAS Build',
       'Creating your first build',
-      'Configuration with eas.json',
+      'EAS Build - Configuration with eas.json',
```
Comment (cruzach) [inline docs/constants/navigation.js:39]:
```suggestion
      'Configuring EAS Build with eas.json',
```

Imo this is a little better of a page title- and fewer characters. And then would say the same change to the EAS Submit title

File: docs/pages/build/eas-json.md [hunk 4]
```diff
@@ -43,7 +42,7 @@ or
...
+The JSON object under the `build` key can contain multiple build profiles. Every build profile can have an arbitrary name. The default profile that is expected by EAS CLI to exist is `release` (if you'd like to build your app using another build profile you need to specify it with a parameter - `eas build --platform android --profile foobar`). In the example, there are two build profiles (`release` and `development`), however they could be named `foo` or `bar` or whatever you'd like. Inside build profile you can specify `android` and `ios` fields that contain platform-specific configuration for the build, any common options can be also stored there or in the root of the build profile.
```
Comment (cruzach) [inline docs/pages/build/eas-json.md:45]:
```suggestion
The JSON object under the `build` key can contain multiple build profiles. Every build profile can have an arbitrary name. The default profile that is expected by EAS CLI to exist is `release` (if you'd like to build your app using another build profile you need to specify it with a parameter - `eas build --platform android --profile foobar`). In the example, there are two build profiles (`release` and `development`), however they could be named `foo` or `bar` or whatever you'd like. Inside each build profile you can specify `android` and `ios` fields that contain platform-specific configuration for the build, any common options can be also stored there or in the root of the build profile.
```

File: docs/pages/submit/android.md [hunk 5]
```diff
@@ -23,42 +24,41 @@ See [expo.fyi/first-android-submission](https://expo.fyi/first-android-submissio
 
 ## 1. Build a standalone app
 
-You'll need a built native app binary. You can either use the [EAS Build](introduction.md) service or do it on your own.
+You'll need a native app binary signed for store submission. You can either use the [EAS Build](introduction.md) service or do it on your own.
```
Comment (cruzach) [inline docs/pages/submit/android.md:27]:
```suggestion
You'll need a native app binary signed for store submission. You can either use [EAS Build](introduction.md) or do it on your own.
```

**Review by cruzach — APPROVED** [review]

**Pull request merged**